instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py

```diff
@@ -0,0 +1,250 @@
+_excluded_urls = get_excluded_urls("AIOHTTP_SERVER")
+
+
+def _parse_duration_attrs(req_attrs):
+    duration_attrs = {}
+    for attr_key in _duration_attrs:
+        if req_attrs.get(attr_key) is not None:
+            duration_attrs[attr_key] = req_attrs[attr_key]
+    return duration_attrs
+
+
+def _parse_active_request_count_attrs(req_attrs):
+    active_requests_count_attrs = {}
+    for attr_key in _active_requests_count_attrs:
+        if req_attrs.get(attr_key) is not None:
+            active_requests_count_attrs[attr_key] = req_attrs[attr_key]
+    return active_requests_count_attrs
+
+
+def get_default_span_details(request: web.Request) -> Tuple[str, dict]:
+    """Default implementation for get_default_span_details
+    Args:
+        scope: the asgi scope dictionary
+    Returns:
+        a tuple of the span name, and any attributes to attach to the span.
+    """
+    span_name = request.path.strip() or f"HTTP {request.method}"
+    return span_name, {}
+
+
+def _get_view_func(request) -> str:
+    """TODO: is this useful??"""
+    try:
+        return request.match_info.handler.__name__
+    except AttributeError:
+        return "unknown"
+
+
+def collect_request_attributes(request: web.Request):
+    """Collects HTTP request attributes from the ASGI scope and returns a
+    dictionary to be used as span creation attributes."""
+
+    server_host, port, http_url = (
+        request.url.host,
+        request.url.port,
+        str(request.url),
+    )
+    query_string = request.query_string
+    if query_string and http_url:
+        if isinstance(query_string, bytes):
+            query_string = query_string.decode("utf8")
+        http_url += "?" + urllib.parse.unquote(query_string)
+
+    result = {
+        SpanAttributes.HTTP_SCHEME: request.scheme,
+        SpanAttributes.HTTP_HOST: server_host,
+        SpanAttributes.NET_HOST_PORT: port,
+        SpanAttributes.HTTP_ROUTE: _get_view_func(request),
+        SpanAttributes.HTTP_FLAVOR: f"{request.version.major}.{request.version.minor}",
+        SpanAttributes.HTTP_TARGET: request.path,
+        SpanAttributes.HTTP_URL: remove_url_credentials(http_url),
+    }
+
+    http_method = request.method
+    if http_method:
+        result[SpanAttributes.HTTP_METHOD] = http_method
+
+    http_host_value_list = (
+        [request.host] if type(request.host) != list else request.host
+    )
+    if http_host_value_list:
+        result[SpanAttributes.HTTP_SERVER_NAME] = ",".join(
+            http_host_value_list
+        )
+    http_user_agent = request.headers.get("user-agent")
+    if http_user_agent:
+        result[SpanAttributes.HTTP_USER_AGENT] = http_user_agent
+
+    # remove None values
+    result = {k: v for k, v in result.items() if v is not None}
+
+    return result
+
+
+def set_status_code(span, status_code):
+    """Adds HTTP response attributes to span using the status_code argument."""
+    if not span.is_recording():
```

Comment:
`set_status` and `set_attribute` are no-ops for non-recording spans. In the event this changes in the future, it is better to call those methods instead of bypassing them. I recommend deleting this `if` statement.